overlord/configstate/config/helpers.go

```diff
@@ -47,6 +47,46 @@ func ParseKey(key string) (subkeys []string, err error) {
 	return subkeys, nil
 }
 
+func purgeNulls(config interface{}) interface{} {
+	switch config := config.(type) {
+	// map of json raw messages is the starting point for purgeNulls, this is the configuration we receive
+	case map[string]*json.RawMessage:
+		for k, v := range config {
+			cfg := purgeNulls(v)
+			if cfg != nil {
+				config[k] = cfg.(*json.RawMessage)
+			} else {
+				delete(config, k)
+			}
+		}
+		// After removing nulls from the map we can end up with an empty map
```

Comment:
this wasn't part of the plan, the plan mentions `null`,  not empty maps

Reply:
Right; I didn't think of them when we discussed the plan, and I wasn't sure if we considered this. When implementing this I found leaving empty maps around slightly confusing. I can drop it.

Reply:
Removed.